test/fakeintake/server/server.go

```diff
@@ -286,21 +312,83 @@ func (fi *Server) safeAppendPayload(route string, data []byte, encoding string)
 	if _, found := fi.payloadStore[route]; !found {
 		fi.payloadStore[route] = []api.Payload{}
 	}
-	fi.payloadStore[route] = append(fi.payloadStore[route], api.Payload{
+	rawPayload := api.Payload{
 		Timestamp: fi.clock.Now(),
 		Data:      data,
 		Encoding:  encoding,
-	})
+	}
+	parsedPayload := api.ParsedPayload{
+		Timestamp: rawPayload.Timestamp,
+		Data:      "",
+		Encoding:  encoding,
+	}
+	fi.payloadStore[route] = append(fi.payloadStore[route], rawPayload)
+
+	if route == "/api/v2/logs" {
+		parsedPayload.Data = fi.getLogPayLoadData(rawPayload)
+	} else if route == "/api/v2/series" {
+		parsedPayload.Data = fi.getMetricPayLoadData(rawPayload)
+	} else if route == "/api/v1/check_run" {
+		parsedPayload.Data = fi.getCheckRunPayLoadData(rawPayload)
+	}
+	fi.payloadJsonStore[route] = append(fi.payloadJsonStore[route], parsedPayload)
+}
+
+func (fi *Server) getLogPayLoadData(payload api.Payload) string {
+	logs, err := aggregator.ParseLogPayload(payload)
+	if err != nil {
+		return err.Error()
+	}
+
+	output, er := json.Marshal(logs)
+	if er != nil {
+		return er.Error()
+	}
+	return string(output)
+}
+
+func (fi *Server) getMetricPayLoadData(payload api.Payload) string {
+	MetricOutput, err := aggregator.ParseMetricSeries(payload)
+	if err != nil {
+		return err.Error()
+	}
+
+	output, er := json.Marshal(MetricOutput)
+	if er != nil {
+		return er.Error()
+	}
+	return string(output)
+}
+
+func (fi *Server) getCheckRunPayLoadData(payload api.Payload) string {
+	MetricOutput, err := aggregator.ParseCheckRunPayload(payload)
+	if err != nil {
+		return err.Error()
+	}
+
+	output, er := json.Marshal(MetricOutput)
+	if er != nil {
+		return er.Error()
+	}
+	return string(output)
 }
 
-func (fi *Server) safeGetPayloads(route string) []api.Payload {
+func (fi *Server) safeGetRawPayloads(route string) []api.Payload {
 	fi.mu.Lock()
 	defer fi.mu.Unlock()
 	payloads := make([]api.Payload, 0, len(fi.payloadStore[route]))
 	payloads = append(payloads, fi.payloadStore[route]...)
 	return payloads
 }
 
+func (fi *Server) safeGetJsonPayloads(route string) []api.ParsedPayload {
```

Comment:
It should return an error if this function is called with a route not covered by https://github.com/DataDog/datadog-agent/pull/18332/files#diff-ff3f92894e768d7fb2774faacacdd9a69fde06f3c3e81bfa1ed81b052204e7eaR327-R331.